test/jest/acceptance/api-auto-detect.spec.ts

```diff
@@ -0,0 +1,27 @@
+import { runSnykCLI } from '../util/runSnykCLI';
+
+jest.setTimeout(1000 * 10);
+
+describe('Auto detect API URL', () => {
+  it('Based on OAuth token claim', async () => {
+    const expectedApiUrl = 'https://api.my.special.url.io';
+    const oauthToken = {
+      access_token:
+        'a.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCJhdWQiOlsiaHR0cHM6Ly9hcGkubXkuc3BlY2lhbC51cmwuaW8iXX0.b',
+      token_type: 'Bearer',
+      refresh_token: 'configRefreshToken',
+      expiry: '3023-03-29T17:47:13.714448+02:00',
+    };
```

Comment:
What do you think about something like this for test readability (didn't test it, just for suggestions)

```suggestion
    const payload = {
      sub: '1234567890',
      name: 'John Doe',
      iat: 1516239022,
      aud: [expectedApiUrl],
    };

    const oauthToken = {
      access_token:
        `a.${Buffer.from(
      JSON.stringify(readableTokenPayload)
    ).toString('base64')}.b`,
      token_type: 'Bearer',
      refresh_token: 'configRefreshToken',
      expiry: '3023-03-29T17:47:13.714448+02:00',
    };
```

Reply:
Awesome idea! Thank you, just needed some small changes to make it work 😄